frontend/packages/operator-lifecycle-manager/src/components/catalog-source.spec.tsx

```diff
@@ -33,47 +26,121 @@ jest.mock('react-router-dom-v5-compat', () => ({
+  Grid: jest.fn(({ children }) => children),
+  GridItem: jest.fn(({ children }) => children),
+  DescriptionList: jest.fn(({ children }) => children),
+}));
+
+const mockDetailsPage = (DetailsPage as unknown) as jest.Mock;
+const mockFirehose = (Firehose as unknown) as jest.Mock;
+const mockUseK8sWatchResource = useK8sWatchResource as jest.Mock;
+
+describe('CatalogSourceDetails', () => {
+  let obj;
 
   beforeEach(() => {
     obj = _.cloneDeep(testCatalogSource);
-    wrapper = shallow(<CatalogSourceDetails obj={obj} packageManifests={[testPackageManifest]} />);
   });
 
-  it('renders name and publisher of the catalog', () => {
-    expect(wrapper.find(DetailsItem).at(1).props().obj.spec.displayName).toEqual(
-      obj.spec.displayName,
+  it('displays catalog source name and publisher', () => {
+    renderWithProviders(
+      <CatalogSourceDetails obj={obj} packageManifests={[testPackageManifest]} />,
     );
 
-    expect(wrapper.find(DetailsItem).at(2).props().obj.spec.publisher).toEqual(obj.spec.publisher);
+    expect(screen.getByText(obj.spec.displayName, { exact: false })).toBeVisible();
+    expect(screen.getByText(obj.spec.publisher, { exact: false })).toBeVisible();
   });
 });
 
-describe(CatalogSourceDetailsPage.displayName, () => {
-  let wrapper: ShallowWrapper;
-
+describe('CatalogSourceDetailsPage', () => {
   beforeEach(() => {
-    (useK8sWatchResource as jest.Mock).mockReturnValue([dummyPackageManifest, true, null]);
+    mockUseK8sWatchResource.mockReturnValue([dummyPackageManifest, true, null]);
     jest.spyOn(Router, 'useParams').mockReturnValue({ ns: 'default', name: 'some-catalog' });
-    wrapper = shallow(<CatalogSourceDetailsPage />);
+    mockDetailsPage.mockClear();
+  });
+
+  it('renders catalog source details page without errors', () => {
+    expect(() => {
+      renderWithProviders(<CatalogSourceDetailsPage />);
+    }).not.toThrow();
   });
 
-  it('renders `DetailsPage` with correct props', () => {
-    expect(wrapper.find(DetailsPage).props().kind).toEqual(referenceForModel(CatalogSourceModel));
+  it('configures DetailsPage with correct navigation and resources', () => {
+    renderWithProviders(<CatalogSourceDetailsPage />);
 
-    const detailsPage = wrapper.find(DetailsPage);
-    const { pages } = detailsPage.props();
-    expect(pages.length).toEqual(3);
-    expect(pages[0].nameKey).toEqual(`public~Details`);
-    expect(pages[1].nameKey).toEqual(`public~YAML`);
-    expect(pages[2].nameKey).toEqual(`olm~Operators`);
+    expect(mockDetailsPage).toHaveBeenCalledTimes(1);
+    const [detailsPageProps] = mockDetailsPage.mock.calls[0];
 
-    expect(pages[0].component).toEqual(CatalogSourceDetails);
-    expect(pages[2].component).toEqual(CatalogSourceOperatorsPage);
+    expect(detailsPageProps.kind).toEqual(referenceForModel(CatalogSourceModel));
 
-    expect(wrapper.find(DetailsPage).props().resources).toEqual([
+    expect(detailsPageProps.pages).toHaveLength(3);
```

Comment:
This tests implementation details (how DetailsPage is called), not user behavior. I would make it a follow-up in the story to extend RTL tests.